libindex/fetcher.go

```diff
@@ -206,14 +209,32 @@ func (a *RemoteFetchArena) realizeLayer(ctx context.Context, l *claircore.Layer)
 	req := &http.Request{
 		ProtoMajor: 1,
 		ProtoMinor: 1,
-		Method:     http.MethodGet,
+		Method:     http.MethodHead,
 		URL:        url,
 		Header:     l.Headers,
 	}
+	headResp, err := a.wc.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("fetcher: HEAD request failed: %w", err)
+	}
+
+	if cl := headResp.Header.Get("content-length"); cl != "" {
+		size, err := strconv.ParseInt(cl, 10, 64)
+		switch {
+		case err == nil:
+			if a.maxBytes > 0 && size > a.maxBytes {
+				return "", fmt.Errorf("fetcher: layer size (%d) larger than maximum layer size (%d)", size, a.maxBytes)
+			}
+		default:
+			zlog.Warn(ctx).Msg("unable to convert content-length header to bytes")
+		}
+	}
+
+	req.Method = http.MethodGet
```

Comment:
This won't work with signed URLs, they usually have only a set of allowed methods.

Reply:
I changed it to just make the GET request which means we should avoid any latency implication as long as the length check comes before the body reading.

Reply:
need to update the commit msg